builtin/credential/aws-ec2/client.go

```diff
@@ -118,7 +118,7 @@ func (b *backend) clientEC2(s logical.Storage, region string, stsRole string) (*
 	defer b.configMutex.Unlock()
 
 	// If the client gets created while switching the locks, return it
-	if b.EC2ClientsMap[region][stsRole] != nil {
+	if b.EC2ClientsMap[region] != nil && b.EC2ClientsMap[region][stsRole] != nil {
```

Comment:
I think the idea here is that we check again if the client has been created whilst the read lock was switched for a read/write lock. Happy to just obtain the read/write lock at the top of the function if you think this is preferable?